pkg/cmd/cluster/cluster_link.go

```diff
@@ -0,0 +1,101 @@
+package cluster
+
+import (
+	"errors"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/clusterlinker"
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/dcos/dcos-cli/pkg/login"
+	"github.com/dcos/dcos-cli/pkg/setup"
+	"github.com/spf13/cobra"
+)
+
+// newCmdClusterLink links the attached cluster to another one.
+func newCmdClusterLink(ctx api.Context) *cobra.Command {
+	setupFlags := setup.NewFlags(ctx.Fs(), ctx.EnvLookup)
+	cmd := &cobra.Command{
+		Use:  "link",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			attachedCluster, err := ctx.Cluster()
+			if err != nil {
+				return err
+			}
+
+			manager := ctx.ConfigManager()
+			linkableClusterConfig, err := manager.Find(args[0], false)
+			if err != nil {
+				if err == config.ErrConfigNotFound {
+					msg := " is not set up in the CLI, would you like to do it now?"
+					err = ctx.Prompt().Confirm(args[0] + msg)
+					if err != nil {
+						return err
+					}
+
+					// We do not want to attach the linkable cluster.
+					attach := false
+					ctx.Setup(setupFlags, args[0], attach)
```

Comment:
This could simply become `ctx.Setup(setupFlags, args[0])` if we leave the attachment responsibility to the caller.